### PR TITLE
cls/queue: always set member variables in ctor

### PR DESCRIPTION
**Pull request by tchaikov**
this should address the test failures like

```
/tmp/typ-WWFeFl6yK /tmp/typ-sMfwoaGMU differ: byte 24, line 1
**** cls_2pc_reservation test 2 binary reencode check failed ****
   ceph-dencoder type cls_2pc_reservation select_test 2 encode export /tmp/typ-WWFeFl6yK
   ceph-dencoder type cls_2pc_reservation select_test 2 encode decode encode export /tmp/typ-sMfwoaGMU
2c2
< 00000010  00 00 00 00 00 00 c0 c6  92 10                    |..........|
---
> 00000010  00 00 00 00 00 00 c0 e6  cd 53                    |.........S|
```

where we
1. encode the 2nd sample created by `generate_test_instances()`,
2. encode, decode, and encode again, the same sample

and compare the encoded blobs.

but if we fail to set any of the fields in `cls_2pc_reservation`, we are at the mercy of the random bits on stack/heap.

in this change, all bits are initialized.

the flaky test was introduced by 1d7cabf3

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/cls/2pc_queue/cls_2pc_queue_types.h [hunk 1]
```diff
@@ -38,6 +38,7 @@ struct cls_2pc_reservation
   void dump(ceph::Formatter *f) const {
     f->dump_unsigned("size", size);
     f->dump_stream("timestamp") << timestamp;
+    f->dump_stream("timestamp") << timestamp;
```
Comment (cbodley) [inline src/cls/2pc_queue/cls_2pc_queue_types.h:41]: unintended change? duplicates the line above

**Review by cbodley — COMMENTED** [review]
thanks @tchaikov

File: src/cls/2pc_queue/cls_2pc_queue_types.h (hunk 1 above)
Comment (tchaikov) [reply src/cls/2pc_queue/cls_2pc_queue_types.h:41]: sorry, Casey. i intended to dump `entries` also. but changed my mind -- thought it should be included in another PR.

**Comment by tchaikov** [comment]
changelog

- drop unintended `f->dump_stream("timestamp") << timestamp;`.

**Comment by tchaikov** [comment]
@cbodley hi Casey, thank you for your review. could you please take another look?

**Review by cbodley — APPROVED** [review]

**Comment by cbodley** [comment]
> The following tests FAILED:
	198 - unittest_osdmap (Subprocess aborted)

**Comment by cbodley** [comment]
> jenkins.util.io.CompositeIOException: Unable to delete '/home/jenkins-build/build/workspace/ceph-pull-requests'. Tried 3 times (of a maximum of 3) waiting 0.1 sec between attempts. (Discarded 11124 additional exceptions)

**Comment by cbodley** [comment]
jenkins test make check

**Comment by cbodley** [comment]
tyvm @tchaikov :heart:

**Pull request merged**